src/frontend/components/SelectContent/index.tsx

```diff
@@ -0,0 +1,337 @@
+    id: 'components.SelectContent.notReadyToShow',
+  },
+  select: {
+    defaultMessage: 'Select {content_title}',
+    description: `Title used for a video or a document select.`,
+    id: 'components.SelectContent.select',
+  },
+  titleEdit: {
+    defaultMessage: 'LTI Content title',
+    description: `Label for LTI title content input.`,
+    id: 'components.SelectContent.titleEdit',
+  },
+});
+
+const THUMBNAIL_SIZE = '144';
+
+const IconBox = styled.span`
+  font-size: 70px;
+  text-align: center;
+  padding: 40px;
+`;
+
+const IconStatus = ({
+  message,
+  GrommetIcon,
+  color,
+}: {
+  message: string;
+  GrommetIcon: Icon;
+  color: string;
+}) => (
+  <Box direction="row" gap="small" pad="small">
+    <GrommetIcon a11yTitle={message} color={color} />
+    <Text>{message}</Text>
+  </Box>
+);
+
+const AssertedIconStatus = ({
+  assertion,
+  trueMessage,
+  falseMessage,
+  TrueIcon,
+  FalseIcon,
+}: {
+  assertion: boolean;
+  trueMessage: string;
+  falseMessage: string;
+  TrueIcon: Icon;
+  FalseIcon: Icon;
+}) => {
+  if (assertion) {
+    return (
+      <IconStatus
+        message={trueMessage}
+        GrommetIcon={TrueIcon}
+        color="status-ok"
+      />
+    );
+  }
+
+  return (
+    <IconStatus
+      message={falseMessage}
+      GrommetIcon={FalseIcon}
+      color="status-error"
+    />
+  );
+};
+
+const ContentCard = ({
+  content,
+  onClick,
+}: {
+  content: Nullable<Video> | Nullable<Document>;
+  onClick: () => void;
+}) => {
+  const intl = useIntl();
+  const contentTitle = { content_title: content?.title };
+
+  return (
+    <Tip
+      content={
+        <Box pad="medium">
+          <Text>{content?.title}</Text>
+          <Box gap="small" direction="row" align="end">
+            <AssertedIconStatus
+              assertion={content?.upload_state === 'ready'}
+              trueMessage={intl.formatMessage(messages.uploaded)}
+              falseMessage={intl.formatMessage(messages.notUploaded)}
+              TrueIcon={DocumentUpload}
+              FalseIcon={DocumentMissing}
+            />
+          </Box>
+
+          <Box gap="small" direction="row" align="end">
+            <AssertedIconStatus
+              assertion={content?.is_ready_to_show!}
+              trueMessage={intl.formatMessage(messages.readyToShow)}
+              falseMessage={intl.formatMessage(messages.notReadyToShow)}
+              TrueIcon={Monitor}
+              FalseIcon={Monitor}
+            />
+          </Box>
+        </Box>
+      }
+    >
+      <Card
+        width="large"
+        title={intl.formatMessage(messages.select, contentTitle)}
+        onClick={onClick}
+      >
+        <CardBody height="small">
+          {'urls' in content! && content?.urls ? (
+            <Image
+              alignSelf="stretch"
+              fit="cover"
+              fill="vertical"
+              src={content?.urls.thumbnails[THUMBNAIL_SIZE]}
```

Comment:
This size may not exists. You don't know the available sizes. For this you have to find the available sizes and take the smaller.

Reply:
Oh ok, thanks

Reply:
I think by default they are ordered from the lower to the higher. Check this point but I think taking the first item in the array is enough.

Reply:
In fact, I have to find the size by another way than url.thumbnails array as I need it for here too https://github.com/openfun/marsha/pull/902/files#diff-a57ea6d3b5fea5302c3087db3f99786eb674ca04bb5b6b72d3d8419dbdded9b9R229
I'll try from `import { videoSize } from '../../types/tracks';`

Reply:
or videoSizeMapping

Reply:
You can not find them from the defined types. In the types you have all possible values. Each video can have different sizes.
IMO keeping `144` for the card size is good and then pick the smaller size available by accessing the value at the index `0`.

Reply:
ok :+1:  thanks again